src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/barnetilsyn/TilsynBarnBeregningService.kt

```diff
@@ -1,11 +1,166 @@
+import java.time.YearMonth
+import java.util.UUID
+
+private val SATS_PROSENT = BigDecimal(0.64)
+private val SNITT_ANTALL_DAGER_PER_MÅNED = BigDecimal(21.67)
 
 @Service
 class TilsynBarnBeregningService {
 
-    fun beregn(dto: InnvilgelseTilsynBarnDto): BeregningsresultatTilsynBarnDto {
-        return BeregningsresultatTilsynBarnDto(emptyList())
+    // Hva burde denne ta inn? Hva burde bli sendt inn i beregningscontroller?
+    fun beregn(
+        stønadsperioder: List<Stønadsperiode>,
+        utgifterPerBarn: Map<UUID, List<Utgift>>,
+    ): BeregningsresultatTilsynBarnDto {
+        validerPerioder(stønadsperioder, utgifterPerBarn)
+
+        val beregningsgrunnlag = lagBeregningsgrunnlag(stønadsperioder, utgifterPerBarn)
+        val perioder = beregn(beregningsgrunnlag)
+
+        return BeregningsresultatTilsynBarnDto(perioder)
+    }
+
+    private fun beregn(beregningsgrunnlag: List<Beregningsgrunnlag>): List<Beregningsresultat> {
+        return beregningsgrunnlag.map {
+            Beregningsresultat(
+                makssats = 100, // TODO trenger makssats
+                dagsats = beregn(it),
+                grunnlag = it,
+            )
+        }
+    }
+
+    private fun beregn(grunnlag: Beregningsgrunnlag): Float {
+        val utgifter = grunnlag.utgifterTotal.toBigDecimal()
+        return utgifter.multiply(SATS_PROSENT)
+            .divide(SNITT_ANTALL_DAGER_PER_MÅNED, 4, RoundingMode.HALF_UP)
+            .setScale(2, RoundingMode.HALF_UP)
+            .toFloat()
+    }
+
+    private fun lagBeregningsgrunnlag(
+        stønadsperioder: List<Stønadsperiode>,
+        utgifterPerBarn: Map<UUID, List<Utgift>>,
+    ): List<Beregningsgrunnlag> {
+        val stønadsdagerPerÅrMåned = stønadsperioder.tilÅrMåneder()
+
+        val stønadsperiode = stønadsperiode(stønadsperioder)
+
+        val utgifterPerMåned = getUtgifterPerMåned(utgifterPerBarn, stønadsperiode)
+
+        return stønadsdagerPerÅrMåned.entries.mapNotNull { (måned, stønadsperioder) ->
+            utgifterPerMåned[måned]?.let { utgifter ->
+                Beregningsgrunnlag(
+                    måned = måned,
+                    stønadsperioder = stønadsperioder,
+                    utgifter = utgifter,
+                    antallDagerTotal = antallDager(stønadsperioder),
+                    utgifterTotal = utgifter.sumOf { it.utgift },
+                )
+            }
+        }
+    }
+
+    private fun antallDager(stønadsperioder: List<Stønadsperiode>): Int {
+        return stønadsperioder.sumOf { stønadsperiode ->
+            stønadsperiode.alleDatoer().count { !VirkedagerProvider.erHelgEllerHelligdag(it) }
+        }
+    }
+
+    private fun stønadsperiode(stønadsperioder: List<Stønadsperiode>): ClosedRange<YearMonth> {
+        val førsteStønadsmåned = YearMonth.from(stønadsperioder.first().fom)
+        val sisteStønadsmåned = YearMonth.from(stønadsperioder.last().tom)
+        return førsteStønadsmåned..sisteStønadsmåned
```

Comment:
Dette ser ut som magi 👀, hvordan funker det? Fylles inn månedene mellom her automatisk av `..`?´ 😲

Reply:
`..` blir en range, sånn at når man kjører 
`dato in fom..tom` så blir det samme som `fom <= dato && dato <= tom`